lib/Byng/Pimcore/Sitemap/Plugin/Installer.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/**
+ * This file is part of the pimcore-sitemap-plugin package.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Byng\Pimcore\Sitemap\Plugin;
+
+use Byng\Pimcore\Sitemap\SitemapPlugin;
+use Pimcore\Config;
+use Pimcore\Model\Property\Predefined as PredefinedProperty;
+use Pimcore\Model\Site;
+use Pimcore\Model\Staticroute;
+
+/**
+ * Class Installer
+ *
+ * @package Byng\Pimcore\Sitemap\Plugin
+ * @author Yann Weyer <yann.weyer@orderbird.com>
+ */
+final class Installer
+{
+    /**
+     * Creates predefined properties to be used in the admin interface
+     */
+    public function createProperties()
+    {
+        $data = [
+            "key" => "sitemap_exclude",
+            "name" => "Sitemap: Exclude page",
+            "description" => "Add this property to exclude a page from the sitemap",
+            "ctype" => "document",
+            "type" => "bool",
+            "inheritable" => false,
+            "data" => true
+        ];
+        $property = PredefinedProperty::create();
+        $property->setValues($data);
+
+        $property->save();
+    }
+
+    /**
+     * Deletes predefined properties
+     */
+    public function deleteProperties()
+    {
+        $property = PredefinedProperty::getByKey("sitemap_exclude");
+        $property->delete();
+    }
+
+    /**
+     * Creates the redirect rules used to display the appropriate sitemap file
+     */
+    public function createRedirectRule()
+    {
+        $route = Staticroute::create();
+        $route->setName('sitemap')
+            ->setPattern('/^\/sitemap\.xml$/')
+            ->setReverse('/sitemap.xml')
+            ->setModule('PimcoreSitemapPlugin')
+            ->setController('sitemap')
+            ->setAction('view')
+            ->save();
+    }
+
+    /**
+     * Removes the redirect rule
+     */
+    public function deleteRedirectRule()
+    {
+        $route = Staticroute::getByName('sitemap');
+        $route->delete();
+    }
+
+    /**
+     * Creates the sitemap folder
+     */
+    public function createSitemapFolder()
+    {
+        // Create sitemap folder (if doesn't exist already)
+        if (!is_dir(SITEMAP_PLUGIN_FOLDER)) {
+            mkdir(SITEMAP_PLUGIN_FOLDER, 0777, true);
+        }
+    }
+
+    /**
+     * Deletes the sitemap folder and its children
+     */
+    public function deleteSitemapFolder()
+    {
+        if (is_dir(SITEMAP_PLUGIN_FOLDER)) {
+            if (!is_dir_empty(SITEMAP_PLUGIN_FOLDER)) {
```

Comment:
I'm getting an error here, saying `is_dir_empty` doesn't exist, causing the uninstall function to fail.